script.R

```diff
@@ -10,6 +10,53 @@ if_error <- function(x, y = NULL) {
 `%||%` <- function(x, y) {
   if (!length(x) || is.null(x)) y else x
 }
+check_if_pkg_available <- function(pkg, ver = NULL) {
+  if (is.null(ver)) {
+    length(available.packages(filter = list(add = TRUE, function(db) db[db[, "Package"] == pkg, ]))) > 0
+  } else {
+    length(
+      available.packages(
+        filter = list(
+          add = TRUE,
+          function(db) {
+            db[db[, "Package"] == pkg & db[, "Version"] == ver, ]
+          }
+        )
+      )
+    ) > 0
+  }
+}
+add_to_minicran <- function(x, minicran_path) {
+  temp_dir <- tempfile()
+  on.exit(unlink(temp_dir))
+  dir.create(temp_dir)
+  new_file_name <- gsub("(.*?_.*)_.*?(\\..*)", "\\1\\2", basename(x))
+  file.copy(x, file.path(temp_dir, new_file_name))
+  miniCRAN::addLocalPackage(gsub("_.*", "", basename(new_file_name)), temp_dir, minicran_path)
+  invisible(NULL)
+}
+add_cache_to_minicran <- function(pkg, version, minicran_path) {
+  i_cache <- pkgcache::pkg_cache_find(package = pkg, version = version, platform = "source")
+  if (nrow(i_cache) == 0) return(invisible(NULL))
+  cli::cli_inform(sprintf("Adding %s to miniCRAN...", pkg))
+  add_to_minicran(i_cache$fullpath[1], minicran_path)
+  invisible(NULL)
+}
+install_and_add_to_minicran <- function(ref, minicran_path) {
+  cli::cli_inform(sprintf("Installing and adding %s to miniCRAN...", ref))
+  x <- pak::pkg_install(ref)
```

Comment:
By default `ask = interactive()` so in GitHub actions scripting mode it will be `FALSE` but depending on circumstances on how this is run I would prefer to specify it directly to `FALSE`. This is for the case where this package was already installed, and we are trying to downgrade or upgrade.
```suggestion
  x <- pak::pkg_install(ref, ask = FALSE)
```